Website/Providers/DataProviders/SqlDataProvider/09.02.00.SqlDataProvider

```diff
@@ -736,6 +736,96 @@ BEGIN
 END; -- Procedure
 GO
 
+/******************************************************************************************************************/
+/***** Fix DNN-10557: Fix xml storage for comment & comment moderation in case of longer than 2000 characters *****/
+/******************************************************************************************************************/
+
+IF  EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'{databaseOwner}[{objectQualifier}Journal_Comment_Save]') AND type in (N'P', N'PC'))
+    DROP PROCEDURE {databaseOwner}[{objectQualifier}Journal_Comment_Save]
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}Journal_Comment_Save]
+	@JournalId int,
+	@CommentId int,
+	@UserId int,
+	@Comment nvarchar(max),
+	@CommentXML xml,
+	@DateUpdated datetime
+AS
+
+DECLARE @cxml xml
+DECLARE @xml xml
+DECLARE @cdataComment nvarchar(max)
+
+IF EXISTS(SELECT * FROM {databaseOwner}[{objectQualifier}Journal_Comments] WHERE JournalId = @JournalId AND CommentId = @CommentId)
+BEGIN
+	IF (LEN(@Comment) < 2000)
+	BEGIN
+		IF @CommentXML IS NOT NULL
+		BEGIN
+			SET @cxml = @CommentXML
+			SET @cxml.modify('delete /root/comment')
```

Comment:
From performance perspective, instead of deleting it here, can we prevent API from sending it in the xml?

Reply:
Also, if comment is the only child node, after deletion it'll be "<root />" in the CommentXML field and the UI will still have problem to display comment.

Reply:
UI is not just looking for some value in CommentXML field but specifically the comment node in CommentXML field. See line 419 in https://github.com/dnnsoftware/Dnn.Evoq.Social/commit/54cc1fc66d4a7a4c89d5f1f2fe70c04177b5a884

Reply:
Here's js code, since commentXml is not null (root element only), it won't show comment, right?
if (that.commentXml()) {
    return $(that.commentXml()).find("comment").text().replace(/^<\!\[CDATA\[|\]\]>$/g, '');
} else {
    return that.comment();
}

Reply:
Oops my bad, missed to commit the changes. Here's the updated code for the UI check
https://github.com/dnnsoftware/Dnn.Evoq.Social/commit/fa952c5e9ab3e3097ef52d18e0ab933a63ff27cc

Reply:
Ye, this works. see my first comment? can you improve the API?

Reply:
API call only sends old comment in xml form if the logged in user is a moderator otherwise it doesn't. Will fix it in the next commit.
Will still keep the stored proc part as it is as a safety net.

Reply:
I'm not a fan of putting business logic in stored proc but sure, let's keep this change for now. Ideally, it should reset commentXml to null if there's no child nodes. :)

Reply:
modified API to reset commentXml to null if there's no child nodes. Removed the logic of deleting comment node from stored procedure.